### PR TITLE
Added asset protection

### LINKED ISSUE #47: Add asset protection
Currently the **repository** does not protect the access to any of the source files. This means that files containing sensitive information are accessible for any client.

For example, the following API call will result into the full content of the requested file.

```http
GET http://repository.example.com:3000/path/to/secretAlgorithm.js
```

While the desired result would be a `404 - Not found`.

**Adding protection**
By default we should make everything protected (this is the safest option), and whitelist files or folders that are publicly accessible. For the whitelist we need to add an additional configuration option to the repository. Something like:

```json
{
    "url": "http://repository.example.com:3000",
    "repository":
    {
        "source": "./src",
        "cache": "./cache",
        "index": "index.html",
        "assets": [ "*.html", "assets/**/*" ]
    }
}
```

In this option we simply specify what the assets are. The given paths are relative to the source folder.

**Note** that components from protected files always need to be placed into a segment in order to make them internally accessible.

### PR DESCRIPTION
**Pull request by BasMasking**
Closes #47 

Changes proposed in this pull request:
- added allow list to the local repository
- added configuration for assets for repository and standalone mode
- updated examples with asset list
- added tests for public and private assets

@petermasking for review

### CODE REVIEW

File: jitar/src/runtime/LocalRepository.ts [hunk 1]
```diff
@@ -63,6 +66,11 @@ export default class LocalRepository extends Repository
 
     async loadAsset(filename: string): Promise<File>
     {
+        if (this.#assets.includes(filename) === false)
+        {
+            throw new FileNotFound(filename);
+        }
+
         return this.#fileManager.load(filename);
```
Comment (petermasking) [inline jitar/src/runtime/LocalRepository.ts:74]: Suggestion for improvement

```ts
return this.#loadFile(filename);
```

File: jitar/src/server/utils/LocalFileManager.ts [hunk 2]
```diff
@@ -90,33 +90,45 @@ export default class LocalFileManager implements FileManager
...
+    {
+        const promises = patterns.map(pattern => this.filterFiles(pattern));
+        const assetFiles = (await Promise.all(promises)).flat();
+
+        return assetFiles
+            .map(filename => this.getRelativeLocation(filename))
+            .filter(filename => this.#isGeneratedFile(filename) === false);
+    }
+
+    async filterFiles(pattern: string): Promise<string[]>
```
Comment (petermasking) [inline jitar/src/server/utils/LocalFileManager.ts:121]: Can this function be private?

File: jitar/src/server/utils/RuntimeConfigurator.ts [hunk 3]
```diff
@@ -46,14 +46,15 @@ export default class RuntimeConfigurator
     {
         const sourceLocation = configuration.source ?? RuntimeDefaults.SOURCE;
         const cacheLocation = configuration.cache ?? RuntimeDefaults.CACHE;
+        const assets = configuration.assets;
```
Comment (petermasking) [inline jitar/src/server/utils/RuntimeConfigurator.ts:49]: Suggestion for a more descriptive name: `assetFilePatterns`

**Review by petermasking — CHANGES_REQUESTED** [review]
Great work! Added some suggestions for improvements.

File: jitar/src/runtime/LocalRepository.ts (hunk 1 above)
Comment (BasMasking) [reply jitar/src/runtime/LocalRepository.ts]: Done

File: jitar/src/server/utils/RuntimeConfigurator.ts (hunk 3 above)
Comment (BasMasking) [reply jitar/src/server/utils/RuntimeConfigurator.ts]: done

File: jitar/src/server/utils/LocalFileManager.ts (hunk 2 above)
Comment (BasMasking) [reply jitar/src/server/utils/LocalFileManager.ts]: done

**Review by petermasking — APPROVED** [review]
Nice!

**Pull request merged**